src/hotspot/share/gc/shared/preservedMarks.cpp

```diff
@@ -92,57 +92,54 @@ void PreservedMarksSet::init(uint num) {
   assert_empty();
 }
 
-class ParRestoreTask : public AbstractGangTask {
-private:
+class RestorePreservedMarksTask : public AbstractGangTask {
   PreservedMarksSet* const _preserved_marks_set;
   SequentialSubTasksDone _sub_tasks;
-  volatile size_t* const _total_size_addr;
+  volatile size_t _total_size;
+#ifdef ASSERT
+  size_t _total_size_before;
```

Comment:
I know `_total_size` is what was used before, but what do you think about changing it to `_total_count` now when we are touching it.

Reply:
Should we rename `size` to `count` all the way through? Also `PreservedMarks` uses `size`, and that is where this comes from. There are likely other places where we use `size` instead of `count` (or `num`, depending on when the code has been written the first time).
Not sure this is the appropriate CR to do that, but not against it either.

Reply:
I decided to keep this kind of renaming out of this CR. We can discuss this offline and file an issue that I can look at.

Reply:
Sorry for missing your reply. Notification got lost, I'm fine with doing this as a separate thing if we feel the need.